### PR TITLE
AG-12369 Fix keynav using initMenuKeyNav and correct aria roles

### PR DESCRIPTION
**Pull request by olegat**
https://ag-grid.atlassian.net/browse/AG-12369

### CODE REVIEW

File: packages/ag-charts-enterprise/src/features/popover/popover.ts [hunk 1]
```diff
@@ -101,63 +98,15 @@ export class Popover extends _ModuleSupport.BaseModuleInstance implements _Modul
...
-            switch (e.key) {
-                case 'ArrowUp':
-                    focusIndexDelta(-1);
-                    break;
-                case 'ArrowDown':
-                    focusIndexDelta(1);
-                    break;
-                case 'Escape':
-                case 'Enter':
-                    opts.onClose();
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/popover/popover.ts:132]: Is `Escape` handling still maintained? I see there is an `onEscape` option to `initMenuKeyNav()` that might need to be used?

File: packages/ag-charts-enterprise/src/features/popover/popover.ts [hunk 2]
```diff
@@ -101,63 +98,15 @@ export class Popover extends _ModuleSupport.BaseModuleInstance implements _Modul
...
-            if (popover.contains(e.target as any)) return;
-            popover.focus();
-        };
-        window.addEventListener('mousemove', windowMouseMoveEvent);
-
-        const windowMouseMoveDown = (e: MouseEvent) => {
-            if (e.button !== 0 || popover.contains(e.target as any)) return;
-
-            opts.onClose();
-        };
```
Comment (alantreadway) [inline packages/ag-charts-enterprise/src/features/popover/popover.ts:152]: Is the behaviour of 'clicks outside the popup close the popup' still maintained?

Comment (olegat) [reply packages/ag-charts-enterprise/src/features/popover/popover.ts:152]: It's working with the Trend Line and Text Annotations menus, but not with the Chart Type menu. I suspect that the Chart Type menu has some other logic that's clashing with.

File: packages/ag-charts-enterprise/src/features/popover/popover.ts (hunk 1 above)
Comment (olegat) [reply packages/ag-charts-enterprise/src/features/popover/popover.ts:132]: We could easily add an `onEscape: () => this.hide` option, but this wouldn't be a complete solution. The `contextMenu.ts` module keeps a reference to HTMLElement that triggered the popup:

https://github.com/ag-grid/ag-charts/blob/c811b8f9c052098c4dec9b94d76ed508a7583a3f/packages/ag-charts-enterprise/src/features/context-menu/contextMenu.ts#L240

`popover.ts` will need to do something similar. I was going to move the `ContextMenu.lastFocus` feature to be part of the `initMenuKeyNav` method, but on second thoughts I could do this in the PR too.

File: packages/ag-charts-enterprise/src/features/popover/popover.ts (hunk 2 above)
Comment (olegat) [reply packages/ag-charts-enterprise/src/features/popover/popover.ts:152]: Fixed: Clicks outside the popup will close the popup.

Notes:
- Right-clicks will also close the popup (previously, only left-clicks closed the pop-up).
- This change has also been to context menus.

File: packages/ag-charts-enterprise/src/features/popover/popover.ts (hunk 1 above)
Comment (olegat) [reply packages/ag-charts-enterprise/src/features/popover/popover.ts:132]: Fixed. The `ContextMenu.lastFocus` logic is now part of the `initMenuKeyNav` function and therefore is used by both context menu and financial charts popovers.

**Review by alantreadway — APPROVED** [review]
LGTM

**Pull request merged**